tests/components/elevenlabs/test_tts.py

```diff
@@ -0,0 +1,270 @@
+    """Set up internal url."""
+    await async_process_ha_core_config(
+        hass, {"internal_url": "http://example.local:8123"}
+    )
+
+
+@pytest.fixture(name="setup")
+async def setup_fixture(
+    hass: HomeAssistant,
+    config_data: dict[str, Any],
+    config_options: dict[str, Any],
+    request: pytest.FixtureRequest,
+    mock_async_client: AsyncMock,
+) -> AsyncMock:
+    """Set up the test environment."""
+    if request.param == "mock_config_entry_setup":
+        await mock_config_entry_setup(hass, config_data, config_options)
+    else:
+        raise RuntimeError("Invalid setup fixture")
+
+    await hass.async_block_till_done()
+    return mock_async_client
+
+
+@pytest.fixture(name="config_data")
+def config_data_fixture() -> dict[str, Any]:
+    """Return config data."""
+    return {}
+
+
+@pytest.fixture(name="config_options")
+def config_options_fixture() -> dict[str, Any]:
+    """Return config options."""
+    return {}
+
+
+async def mock_config_entry_setup(
+    hass: HomeAssistant, config_data: dict[str, Any], config_options: dict[str, Any]
+) -> None:
+    """Mock config entry setup."""
+    default_config_data = {
+        CONF_API_KEY: "api_key",
+    }
+    default_config_options = {
+        CONF_VOICE: "voice1",
+        CONF_MODEL: "model1",
+    }
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=default_config_data | config_data,
+        options=default_config_options | config_options,
+    )
+    config_entry.add_to_hass(hass)
+    client_mock = AsyncMock()
+    client_mock.voices.get_all.return_value = GetVoicesResponse(voices=MOCK_VOICES)
+    client_mock.models.get_all.return_value = MOCK_MODELS
+    with patch(
+        "homeassistant.components.elevenlabs.AsyncElevenLabs", return_value=client_mock
+    ):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+
+
+@pytest.mark.parametrize(
+    "config_data",
+    [
+        {},
+        {tts.CONF_LANG: "de"},
+        {tts.CONF_LANG: "en"},
+        {tts.CONF_LANG: "ja"},
+        {tts.CONF_LANG: "es"},
+    ],
+)
+@pytest.mark.parametrize(
+    ("setup", "tts_service", "service_data"),
+    [
+        (
+            "mock_config_entry_setup",
+            "speak",
+            {
+                ATTR_ENTITY_ID: "tts.mock_title",
+                tts.ATTR_MEDIA_PLAYER_ENTITY_ID: "media_player.something",
+                tts.ATTR_MESSAGE: "There is a person at the front door.",
+                tts.ATTR_OPTIONS: {tts.ATTR_VOICE: "voice2"},
+            },
+        ),
+    ],
+    indirect=["setup"],
+)
+async def test_tts_service_speak(
+    setup: AsyncMock,
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    calls: list[ServiceCall],
+    tts_service: str,
+    service_data: dict[str, Any],
+) -> None:
+    """Test tts service."""
+    tts_entity = hass.data[tts.DOMAIN].get_entity(service_data[ATTR_ENTITY_ID])
+    tts_entity._client.generate.reset_mock()
```

Comment:
We don't need to access the entity to access the client since we've patched the client. Use the client patch mock to assert things about the client.